bokehjs/src/coffee/models/renderers/glyph_renderer.coffee

```diff
@@ -157,7 +157,7 @@ export class GlyphRendererView extends RendererView
       else if selected['1d'].indices.length > 0
         selected = selected['1d'].indices
       else
-        selected = []
+        selected = (parseInt(i) for i in Object.keys(selected["2d"].indices))
```

Comment:
Because we're dealing with JS objects acting as maps. E.g. `{1: []}` has one key of value `"1"`.